### DISCUSSION TITLE
Unable to dynamically load fonts from a web source

### DISCUSSION THREAD

**Discussion by greyhoundgames** [SixLabors/Fonts#179] [Q&A]
I am trying to use this library in AWS lambda and loading the font in dynamically from the web. I cannot get past the setup as it errors with Arial could not be found: FontFamilyNotFoundException. I am printing out the font name and family and it correctly has loaded the font as Arial. The examples don't show having to store these fonts somewhere to be used so I assume they are being cached locally. Any input into what I might be doing wrong would be great.
```csharp
byte[] data = await ContentLoader.LoadDataFile("Fonts/arial.ttf",null);
MemoryStream ms = new MemoryStream(data);
FontCollection` fonts = new FontCollection();
FontFamily fm = fonts.Install(ms,out FontDescription fd);
Font font = fm.CreateFont(10, FontStyle.Regular);
context.Logger.Log("Font:" + font.Name+"-"+font.Family);

//Exception triggers in this section
var slc = new SixLaborsCaptchaModule(new SixLaborsCaptchaOptions
{
  DrawLines = 7,
  TextColor = new Color[] { Color.Blue, Color.Black },
  FontFamilies = new string[] {"Arial"},
  FontStyle = FontStyle.Regular,
  FontSize = 10
});
 ```

**Comment by tocsoft** [comment]
sounds like its going to be an issue inside what ever `SixLaborsCaptchaModule` is (that isn't something that our code) ... at no point are you passing your new `FontCollection` (`fonts`) into that module so there would be no way for it to know anything about it. I would have to assume its trying to use the `SystemFonts` rather than the the collection you created and Arial isn't installed in AWS.

**Reply by greyhoundgames** [reply]
Yeah that is what I was wondering if it was caching it in some known place to be used by the other code.

**Reply by JimBobSquarePants** [reply]
https://github.com/aliasadidev/SixLaborsCaptcha/blob/7b2d58c4db2bbe736f81eac551239ef0b0a1541e/src/SixLaborsCaptcha.Core/SixLaborsCaptchaModule.cs#L30-L31

**Reply by JimBobSquarePants** [reply]
I think you should probably be raising this question on their repository since the code you are using is owned by them, not us.

In fact, there's literally a disclaimer in their readme. 

>Characters fonts, default is "Arial", "Verdana", "Times New Roman"
Notice: This default fonts working only on Windows, if you want to run it on Linux you should use the Linux fonts

https://github.com/aliasadidev/SixLaborsCaptcha#sixlaborscaptchaoptions
